client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -284,6 +287,36 @@ private long getRemainingBytes(SegmentMetadataClientFactory metaFactory, StreamC
         return cuts;
     }
 
+    @Override
+    public CompletableFuture<Map<Stream, StreamCut>> getCurrentStreamCut(ScheduledExecutorService backgroundExecutor) {
+        UUID id = UUID.randomUUID();
+        synchronizer.updateStateUnconditionally(new StartStreamCutBarrier(id.toString()));
+        AtomicBoolean fetchStreamCutPending = new AtomicBoolean(true);
+
+        return Futures.loop(fetchStreamCutPending::get, () -> {
```

Comment:
This never times out, yes? It is possible that a given stream cut goes on forever in the case at least one reader never joins, we won't try to fill in. Is this the current behavior we are proposing?